### ISSUE TITLE
disagree with lack of upgrade doc

### ISSUE THREAD

**Issue by abunur** [malcommac/SwiftDate#89]
It might just be that SwiftDate 2.02 is still too raw to use - I had to revert to 1.2 after trying both Master and Develop branches of 2.02.  But, not having any migration documentation seems like a bit of a cop-out.  I see there are several updates in develop branch for documentation, which is great.

Things like NSDate.today() in SwiftDate 1.2 no longer compile in SwiftDate 2.0.2 - we should be documenting what the new format is.  Similarly for someDate.toISOString().

Currently the sample code in Master branch does not work with SwiftDate 2.0.2 regardless of Master or Develop branch being used, though.

Specifically I could not get the toString() code to compile:

// Create a NSDate (UTC) from string and transform it into DateInRegion
let date = "2015-01-05T22:10:55.000Z".toDate(DateFormat.ISO8601)!.inRegion(romeRegion)
// The same behaviour can be also accomplished using DateInRegion(fromString:format:region)
let date = DateInRegion("2015-01-05T22:10:55.000Z",DateFormat,ISO8601,romeRegion)

// We can print the date (it also works with plain NSDate too!)
// Just some examples:
print("\(date.toISO8601String())") // to UTC ISO8601: "2015-01-05T22:10:55.000Z"
print("\(date.toMediumString())") // to E/Rome: "Jan 5, 2015, 11:10:55 PM"
print(date.toString(DateFormat.Custom("YYYY-MM-dd HH 'at' HH:mm"))) // to E/Rome: "2015-01-05 23 at 23:10"

In the above examples, compiler does not know what "romeRegion" is.  We need some updated documentation on how to use the Region elements.  I tried .Rome, .romeRegion and several other variations but none of them are recognized.

Also the method date.toISO8601String() is not recognized.

**Comment by remotevision** [comment]
I have struggled a little with the migration as well. Opted to sift through code commits to see specifically what has changed. Ultimately I have everything working again except for ```let start = NSDate().startOf(.Day, inRegion: Region.defaultRegion())```. The doc suggests you can not provide a ```Region``` but it is not optional.  I get an error of ```Type 'Region' has no member 'defaultRegion'``` but I don't understand why. Any suggestions?

For the others I did this:
```
sunrise.inDefaultRegion().toShortString(false, time: true)
startDate.toString(DateFormat.ISO8601)
```

**Comment by Hout** [comment]
I will try to create some with conversion docs this weekend.

**Comment by Hout** [comment]
Before diving into the documentation, we need to get more consistency in the library. I will raise an enhancement issue to extend NSDate with as many as possible features from DateInRegion in such a way that it behaves like the latter in the default region.

**Comment by efremidze** [comment]
👍

**Comment by Hout** [comment]
Fixed in v3.0.0
